tests/ottest/fixtures/blockchain/Restart.cpp

```diff
@@ -0,0 +1,264 @@
+        amount_in_transaction_);
+
+    begin += blocks_number_;
+    targetHeight += static_cast<int>(MaturationInterval()) + 1;
+
+    auto scan_listener_external_alice_f = scan_listener_alice->get_future(
+        GetHDAccount(user_alice), bca::Subchain::External, targetHeight);
+    auto scan_listener_internal_alice_f = scan_listener_alice->get_future(
+        GetHDAccount(user_alice), bca::Subchain::Internal, targetHeight);
+    auto scan_listener_external_bob_f = scan_listener_bob->get_future(
+        GetHDAccount(user_bob), bca::Subchain::External, targetHeight);
+    auto scan_listener_internal_bob_f = scan_listener_bob->get_future(
+        GetHDAccount(user_bob), bca::Subchain::Internal, targetHeight);
+
+    std::cout << "Mine targetHeight: " << targetHeight << std::endl;
+    // mine MaturationInterval number block with
+    MineBlocks(begin, static_cast<int>(MaturationInterval()) + 1);
+
+    EXPECT_TRUE(scan_listener_alice->wait(scan_listener_external_alice_f));
+    EXPECT_TRUE(scan_listener_alice->wait(scan_listener_internal_alice_f));
+
+    EXPECT_EQ(
+        GetBalance(user_alice),
+        amount_in_transaction_ * blocks_number_ * transaction_in_block_);
+
+    EXPECT_TRUE(scan_listener_bob->wait(scan_listener_external_bob_f));
+    EXPECT_TRUE(scan_listener_bob->wait(scan_listener_internal_bob_f));
+    EXPECT_EQ(
+        GetBalance(user_bob),
+        amount_in_transaction_ * blocks_number_ * transaction_in_block_);
+    EXPECT_EQ(GetTransactions(user_bob).size(), 1);
+    EXPECT_EQ(GetTransactions(user_alice).size(), 1);
+}
+
+void Restart_fixture::mineTransaction(
+    const User& user,
+    const opentxs::blockchain::block::pTxid& transactionToConfirm)
+{
+    std::vector<Transaction> vec;
+
+    auto send_transaction =
+        user.api_->Crypto().Blockchain().LoadTransactionBitcoin(
+            transactionToConfirm);
+    vec.emplace_back(std::move(send_transaction));
+
+    Mine(targetHeight++, vec.size(), default_, vec);
+}
+
+void Restart_fixture::sendCoins(const User& receiver, const User& sender)
+{
+    const auto& network =
+        sender.api_->Network().Blockchain().GetChain(test_chain_);
+
+    const ot::UnallocatedCString memo_outgoing = "test";
+
+    const auto address = GetNextBlockchainAddress(receiver);
+
+    std::cout << sender.name_ + " send " << coin_to_send_ << " to "
+              << receiver.name_ << " address: " << address << std::endl;
+
+    auto future = network.SendToAddress(
+        sender.nym_id_, address, coin_to_send_, memo_outgoing);
+
+    // We need to confirm send transaction, so it is available after restore
+    mineTransaction(sender, future.get().second);
+}
+
+void Restart_fixture::collectOutputs(
+    const User& user_bob,
+    const User& user_alice,
+    Vector<UTXO>& all_bobs_outputs,
+    Vector<UTXO>& all_alice_outputs,
+    std::map<TxoState, std::size_t>& number_of_outputs_per_type_bob,
+    std::map<TxoState, std::size_t>& number_of_outputs_per_type_alice)
```

Comment:
i would prefer to see more reusable method signatures. 
`collectOutputs` should work for a single user.

```
collectOutputs(
    const User& user,
    Vector<UTXO>& all_outputs,
    std::map<TxoState, std::size_t>& number_of_outputs_per_type)
```

and just call it twice in the test, once per user